### DISCUSSION TITLE
Alternative way to use Yolov5 detections?

### DISCUSSION THREAD

**Discussion by NielsVdk92** [ultralytics/yolov5#5171] [Q&A] [answered]
Hello there!

I'm using **YoloV5** for a **home surveillance system** (detection of people entering at a gate). So far I've been **editing** the **detect.py file to fit all my code**, but when wanting to add **more logic** it is just getting **too complicated** (logic such as detection over multiple cameras, multiple regions of detections per camera, ANPR, ...).

So I was thinking that I am **not following a good approach?**
Instead of using detect.py as a base file, I have seen the PyTorch Hub option (tried it, however failing to even make torch.hub.load work succesfully), but I have given up on that for now.

Apart from Pytorch Hub, is there any other alternative that I am not aware of?
Note: I am a total newbie at this so I could be totally missing out on a possibility that a script can send data / frames to detect.py. All progress has been done by watching 100's of youtube videos.

Thanks a lot!
Niels

**Comment by glenn-jocher** [comment, marked as answer]
@NielsVdk92 yes the two main development options are to start from detect.py or to create your own customized python using PyTorch Hub models. Hub inference is simple and allows for unlimited customization. See YOLOv5 PyTorch Hub tutorial to get started:

## YOLOv5 Tutorials

* [Train Custom Data](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://github.com/ultralytics/yolov5/wiki/Tips-for-Best-Training-Results)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://github.com/ultralytics/yolov5/issues/2518)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://github.com/ultralytics/yolov5/issues/475)
* [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36)&nbsp; ⭐ NEW
* [TorchScript, ONNX, CoreML Export](https://github.com/ultralytics/yolov5/issues/251) 🚀
* [Test-Time Augmentation (TTA)](https://github.com/ultralytics/yolov5/issues/303)
* [Model Ensembling](https://github.com/ultralytics/yolov5/issues/318)
* [Model Pruning/Sparsity](https://github.com/ultralytics/yolov5/issues/304)
* [Hyperparameter Evolution](https://github.com/ultralytics/yolov5/issues/607)
* [Transfer Learning with Frozen Layers](https://github.com/ultralytics/yolov5/issues/1314)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

**Reply by NielsVdk92** [reply]
Alright. Then I'll have to keep testing PyTorch Hub. Thanks again!

**Comment by NielsVdk92** [comment]
FYI: On youtube there is a video showing an alternative way of using YoloV5.
https://www.youtube.com/watch?v=QwflgWURfvI&ab_channel=DeepLearning

Haven't tried it, just recently came across it.

Kind regards
